packages/typegpu/tests/vertexLayout.test.ts

```diff
@@ -301,4 +326,102 @@ describe('connectAttributesToShader', () => {
       usedVertexLayouts: [alphaBetaLayout, gammaLayout],
     });
   });
+
+  it('connects a single vec4h attribute', () => {
+    const shaderInputLayout = d.vec4h;
+    const layout = tgpu.vertexLayout((n) => d.looseArrayOf(d.float16x4, n));
+    const attrib = layout.attrib;
+
+    expect(connectAttributesToShader(shaderInputLayout, attrib)).toEqual({
+      bufferDefinitions: [
+        {
+          arrayStride: 8,
+          stepMode: 'vertex',
+          attributes: [
+            {
+              format: 'float16x4',
+              offset: 0,
+              shaderLocation: 0,
+            },
+          ],
+        },
+      ],
+      usedVertexLayouts: [layout],
+    });
+  });
+
+  it('connects a record of attributes from a single layout (with f16 variants)', () => {
+    const shaderInputLayout = {
+      a: d.f16,
+      b: d.location(3, d.vec2h),
+      c: d.u32 /* should get @location(4) automatically */,
+      d: d.f32,
+      e: d.f16,
+    };
+
+    const layout = tgpu.vertexLayout((n) =>
+      d.looseArrayOf(
+        d.looseStruct({
+          alpha: d.f16, // 2 bytes
+          beta: d.float16x2, // 4 bytes
+          gamma: d.u32, // 4 bytes
+          delta: d.float16, // 2 bytes - ironically, this is interpreted as an f32
+          epsilon: d.float16, // 2bytes this is an f16
```

Comment:
```suggestion
          epsilon: d.float16, // 2 bytes - this is an f16
```

Reply:
isn't this also f32 like delta?

Reply:
sure is 👍 this test is from before the removal of `h-attributes` - will fix